src/error.rs

```diff
@@ -182,3 +186,9 @@ impl From<rocksdb::Error> for Error {
         Self::RocksDB(e)
     }
 }
+
+impl<T> From<PoisonError<T>> for Error {
+    fn from(_: PoisonError<T>) -> Self {
+        Self::PoisionError(())
```

Comment:
Small nit, but I think you should be fine with using `PoisonError` here instead of `PoisionError`.

This shouldn't cause any name collisions, but if you're afraid it makes for less clarity you can always add the more explicit module name.